syntheseus/tests/search/algorithms/test_pdvn.py

```diff
@@ -0,0 +1,428 @@
+        and the tree should have the following structure:
+
+                 ----------CC --------
+                /          |          \
+              C            CO         CS
+        """
+
+        output_graph = self.run_alg_for_n_iterations(
+            retrosynthesis_task5,
+            1,
+            policy=policy,
+            value_function_syn=mol_syn_estimate,
+            value_function_cost=mol_cost_estimate,
+        )
+        assert output_graph.reaction_smiles_counter() == Counter(rxns_step1)  # type: ignore[attr-defined]
+
+        # Various checks on root node
+        assert (
+            output_graph.root_node.num_visit == 2
+        )  # 1 pseudo-visit at the beginning, then 1 real visit
+        assert math.isclose(
+            output_graph.root_node.data["pdvn_mcts_v_syn"],
+            0.4,
+        ), "v_syn should be 0.4 average(0, 0.8)"
+        assert math.isclose(
+            output_graph.root_node.data["pdvn_mcts_v_cost"],
+            1.5,
+        ), "v_cost should be 1.5 average(0, 3)"
+
+        # Training data
+        training_data = pdvn_extract_training_data(output_graph)  # type: ignore[arg-type]  # doesn't understand it is AND/OR graph
+        assert all(v == 0 for v in training_data.mol_to_synthesizability.values())  # nothing solved
+        assert (
+            len(training_data.mol_to_min_syn_cost)
+            == len(training_data.mol_to_reactions_for_min_syn)
+            == 0
+        ), "no solutions found means no cost / reaction data"
+
+    def test_by_hand_step2(
+        self,
+        retrosynthesis_task5: RetrosynthesisTask,
+        policy: DictRxnCost,
+        mol_syn_estimate: DictMolCost,
+        mol_cost_estimate: DictMolCost,
+    ) -> None:
+        r"""
+        Continuation of test above.
+
+        In the second iteration the value function for CS should be evaluated,
+        and in the third iteration CO should be expanded and the OO node visited.
+
+        The tree should have the following structure:
+                 ----------CC --------
+                /          |          \
+              C            CO         CS
+                           |
+                -----------------------
+               /       /        \      \
+             C+O      OO         OS     CS
+        """
+
+        output_graph = self.run_alg_for_n_iterations(
+            retrosynthesis_task5,
+            3,
+            policy=policy,
+            value_function_syn=mol_syn_estimate,
+            value_function_cost=mol_cost_estimate,
+        )
+        assert output_graph.reaction_smiles_counter() == Counter(rxns_step2)  # type: ignore[attr-defined]
+
+        # Various checks on root node
+        assert output_graph.root_node.num_visit == 4
+        assert math.isclose(
+            output_graph.root_node.data["pdvn_mcts_v_syn"],
+            2.6 / 4,
+        ), "v_syn should be 0.4 average(0, 0.8, 0.8, 1.0)"
```

Comment:
Value in the error message doesn't match what is being checked
```suggestion
        assert math.isclose(
            output_graph.root_node.data["pdvn_mcts_v_syn"],
            2.6 / 4,
        ), "v_syn should be 0.65 average(0, 0.8, 0.8, 1.0)"
```